### PR TITLE
fixed environment variables error

### PR DESCRIPTION
**Pull request by p2227**
environment variables always are strings.
The string 'false' means true.
`process.env.npm_config_strict_ssl !== 'false'` is also ok.
and , the proxy setting should pass to `caw` not `got`

I test it.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -125,7 +123,15 @@ Download.prototype.run = function (cb) {
 		if (protocol) {
 			protocol = protocol.slice(0, -1);
 		}
-		var agent = caw(this.opts.proxy, {protocol: protocol});
+		var agent;
+		if(conf['strict-ssl'] === false){
```
Comment (p2227) [inline index.js]: `process.env.npm_config_strict_ssl === 'false'`  should works

**Comment by shinnn** [comment]
> I test it.

No. Test are failing. https://travis-ci.org/kevva/download/builds/129680123

File: index.js [hunk 2]
```diff
@@ -125,7 +123,15 @@ Download.prototype.run = function (cb) {
...
+			agent = caw(objectAssign({
+				protocol: protocol,
+				rejectUnauthorized:false
```
Comment (shinnn) [inline index.js]: As my understanding `rejectUnauthorized` is not the one of `agent` properties. It should be directly passed to `https.request`.  https://nodejs.org/api/https.html#https_https_request_options_callback

**Comment by p2227** [comment]
I'm sorry about the code rules.
you should not merge it directly.
but environment variables are  always strings.

**Comment by shinnn** [comment]
Good point. I just updated the original PR. https://github.com/kevva/download/commit/46196e64303b87595c5492d27fb51742002b7eec

**Comment by p2227** [comment]
ok.I read this doc: https://nodejs.org/api/https.html#https_https_request_options_callback
but I actually find out that it works by setting `rejectUnauthorized` `false` into `agent `

we should make a test case to prove it.

**Comment by shinnn** [comment]
The test case is already included in the upstream project: https://github.com/sindresorhus/got/blob/b09afd0ce52d152bddf5b5297f6cd110177ffaea/test/https.js#L42

In that test `rejectUnauthorized` option is actually passed to `http.request`, not to `agent`.

**Comment by p2227** [comment]
may be i just report it as a bug.
my env
`win7x64`
my step
```
npm i kevva/download#46196e6
set npm_config_strict_ssl = false
npm i gulp-imagemin
```
I get the error
`unable to verify the first certificate`

and here is my proposal: the error may come from https://github.com/request/tunnel-agent/blob/a2c2c2acddd86dd6a894cf4ae53e696e796298c9/index.js#L134

**Comment by shinnn** [comment]
I can't fully understand what you tried to describe in https://github.com/kevva/download/pull/101#issuecomment-218935459.

**Comment by kevva** [comment]
Fixed in https://github.com/kevva/download/commit/559a6f5d576a9f8e3e929f2ae83f19cd14379ad9.

**Pull request closed without merging**